### ISSUE TITLE
New field `TaskId.attemptNumber` to capture task retries on Presto

### ISSUE THREAD

**Issue by shrinidhijoshi** [prestodb/presto#19672]
Creating this issue to track enhancing the `TaskId` class with a new field `attemptNumber`. 

The `attemptNumber` field  is particularly useful in the case of presto-on-spark-native, where the java process creates a Task on the cpp process. 

Currently, due to the lack of this field, 2 retries of the same task will have indistinguishable TaskId.
- Affects Query Execution : We are currently avoiding this issue because we cleanup the Task from CPP worker before starting a new task.
- Affects Statistics Collection: Any Task based statistics collection will be affected as all the statistics of retries will be logged against the same TaskId

Checklist 

- [ ] #### Design : 
    - [ ]  Presto tasks will always have this set to `0` 
- [ ] #### Implementation / Testing
    - [ ] https://github.com/prestodb/presto/pull/19649  
    - [ ] Integration testing
- [ ] #### Other Due Diligence 
    - [ ] Could this be a breaking change for any external systems ? (Monitoring systems, etc ?)

**Comment by shrinidhijoshi** [comment]
@rschlussel  @mbasmanova @vermapratyush
